src/components/ChatEntry.js

```diff
@@ -1,22 +1,48 @@
 import React from 'react';
 import './ChatEntry.css';
 import PropTypes from 'prop-types';
+import TimeStamp from './TimeStamp';
 
-const ChatEntry = (props) => {
-  return (
-    <div className="chat-entry local">
-      <h2 className="entry-name">Replace with name of sender</h2>
-      <section className="entry-bubble">
-        <p>Replace with body of ChatEntry</p>
-        <p className="entry-time">Replace with TimeStamp component</p>
-        <button className="like">🤍</button>
-      </section>
-    </div>
-  );
-};
+const ChatEntry = ({id, sender, body, timeStamp, liked, onUpdate }) => {
+  const messageColor =
+    sender === 'Vladimir' ? 'chat-entry local' : 'chat-entry remote';
+  const changeText = liked ? '❤️' : '🤍';
+
+  const LikeButtonClick = () => {
+      const updatedMessage = {
+        id: id,
+        sender: sender,
+        body: body,
+        timeStamp: timeStamp,
+        liked: !liked,
+      };
+      onUpdate(updatedMessage);
+    }
+
+    return (
+      <div className={messageColor}>
+        <h2 className="entry-name">{sender}</h2>
+        <section className="entry-bubble">
+          <p>{body}</p>
+          <p className="entry-time">
+            <TimeStamp time={timeStamp} />
+          </p>
+          <button className="like" onClick={LikeButtonClick}>
+            {changeText}
+          </button>
+        </section>
+      </div>
+    );
+  };
 
 ChatEntry.propTypes = {
-  //Fill with correct proptypes
-};
+    id: PropTypes.number.isRequired,
+  sender: PropTypes.string.isRequired,
+    body: PropTypes.string.isRequired,
+  time: PropTypes.string.isRequired,
```

Comment:
Please make sure your prop types match what you are getting. Here, the prop types says there should be a prop named `time`, but the in ChatLog we send in a prop called `timeStamp`. This causes a prop type warning to pop up.